switchbot/__init__.py

```diff
@@ -132,12 +145,27 @@ def press(self) -> bool:
 class SwitchbotCurtain(SwitchbotDevice):
     """Representation of a Switchbot Curtain."""
 
+    def __init__(self, *args, **kwargs):
+        """Constructor for a Switchbot Curtain.
+        The position of the curtain is saved is self._pos with 0 = open and 100 = closed.
+        This is independent of the calibration of the curtain bot (Open left to right/
+        Open right to left/Open from the middle).
+        The parameter 'reverse_mode' reverse these values, so that 0 = close and 100 = open.
```

Comment:
Yes, in the app/reading from the device 0 = open and 100 = close independently of the calibration.

Reply:
This will be a breaking change, and the opposite of how it is defined in Home Assistant

@daenny  Any comments on this?

Reply:
I have already updated the Home Assistant integration from @alextud [https://github.com/Jan108/ha-switchbot-curtain/tree/feat/reading_state](url). 
In the integration you could define when a cover is closed, if a cover isn't closed it's open (as seen in [https://github.com/home-assistant/core/blob/8bdcdfb8e67d36b17b1839fa0298f3b874ef375d/homeassistant/components/cover/__init__.py#L185](url)). So the reversed position isn't a problem here.
But in Home Assistant's frontend the controls are disabled, if a cover is fullyopen (100) or fullyclosed (0) (so it would disable the wrong button). I have fixed this i my lastest commit.

Reply:
I have already updated the Home Assistant integration from @alextud [https://github.com/Jan108/ha-switchbot-curtain/tree/feat/reading_state](url). 
In the integration you could define when a cover is closed, if a cover isn't closed it's open (as seen in [https://github.com/home-assistant/core/blob/8bdcdfb8e67d36b17b1839fa0298f3b874ef375d/homeassistant/components/cover/__init__.py#L185](url)). So the reversed position isn't a problem here.
But in Home Assistant's frontend the controls are disabled, if a cover is fullyopen (100) or fullyclosed (0) (so it would disable the wrong button). I have fixed this i my lastest commit.

Reply:
@Jan108 What is your thought to do it opposite? (The same definition as in HA, and we would avoid a breaking change here)

reverse_mode = False:
self._pos with o = close and 100 = open

Reply:
My inital thought with the reverse_mode was the possibility to display the position (0-100 Open left to right, 100-0 Open right to left) in the same slider in HA with left is always left (move slider to the left curtain moves to the left, independently of calibration). This works now fine for me. So I wouldn't change the definition. I would rather change the reverse_mode default to True, so it would work default with HA.

Reply:
Thanks for letting me know. If I remember correctly, there was a mismatch with HA and how switchbot reports the state. 
Seeing at: 
https://github.com/home-assistant/core/blob/dev/homeassistant/components/cover/__init__.py#L174
```       
 None is unknown, 0 is closed, 100 is fully open.
```
I would opt to keep that convention as default. I was using the mqtt implementation (https://github.com/fphammerle/switchbot-mqtt) so it is nice if that works "out of the box" without needing to remap the state in the config.

Reply:
I have changed the default value of 'reverse_mode' to True, so that it would work with HA out of the box. 
I think that would be the best solution because now we have the definition of HA as default and the definition of Switchbot is possible if you change 'reverse_mode' in the initation of an object.